x-pack/libbeat/tests/integration/collector_test.go

```diff
@@ -0,0 +1,177 @@
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/exporter/debugexporter"
+	"go.opentelemetry.io/collector/otelcol"
+
+	"github.com/elastic/beats/v7/libbeat/otelbeat/beatconverter"
+	"github.com/elastic/beats/v7/libbeat/otelbeat/providers/fbprovider"
+	"github.com/elastic/beats/v7/libbeat/tests/integration"
+	"github.com/elastic/beats/v7/x-pack/filebeat/fbreceiver"
+)
+
+var schemeMap = map[string]string{
+	"filebeat": "fb",
+}
+
+type TestCollector struct {
+	t          *testing.T
+	tempDir    string
+	otelcol    *otelcol.Collector
+	wg         sync.WaitGroup
+	configFile string
+	beatname   string
+}
+
+// NewTestCollector configures and returns an otel collector intended for testing only
+// It accepts beatname and configuration
+func NewTestCollector(t *testing.T, beatname string, config string) (*TestCollector, error) {
+
+	// create a temp dir
+	tempDir := integration.CreateTempDir(t)
+	// stdoutFile, err := os.Create(filepath.Join(tempDir, "stdout"))
+	// require.NoError(t, err, "error creating stdout file")
+	// stderrFile, err := os.Create(filepath.Join(tempDir, "stderr"))
+	// require.NoError(t, err, "error creating stderr file")
+
+	// create a config file
+	configFile := filepath.Join(tempDir, beatname+".yml")
+	// write configuration to a file
+	if err := os.WriteFile(configFile, []byte(config), 0o644); err != nil {
+		t.Fatalf("cannot create config file '%s': %s", configFile, err)
+	}
+	// adds scheme name as prefix to the configfile
+	beatCfg := schemeMap[beatname] + ":" + configFile
+	// get collector settings
+	set := getCollectorSettings(beatCfg)
+	// get new collector instance
+	otelcol, err := otelcol.NewCollector(set)
+
+	return &TestCollector{
+		t:          t,
+		tempDir:    tempDir,
+		otelcol:    otelcol,
+		wg:         sync.WaitGroup{},
+		configFile: configFile,
+		beatname:   beatname,
+	}, err
+}
+
+// NewTestStartCollector configures and returns an otel collector intended for testing only
+// It accepts beatname and configuration
+func NewTestStartCollector(t *testing.T, beatname string, config string) (*TestCollector, error) {
+	otelcol, err := NewTestCollector(t, beatname, config)
+	if err != nil {
+		return nil, err
+	}
+	err = otelcol.Run()
+	if err != nil {
+		return nil, err
+	}
+
+	t.Cleanup(func() {
+		otelcol.Shutdown()
+		if !t.Failed() {
+			return
+		}
+	})
+
+	return otelcol, err
+}
+
+func (c *TestCollector) ReloadConfig(config string) error {
+	// If collector is started, shut it down then reload
+	c.Shutdown()
+	// write configuration to a file
+	if err := os.WriteFile(c.configFile, []byte(config), 0o644); err != nil {
+		c.t.Fatalf("cannot create config file '%s': %s", c.configFile, err)
+	}
+	// adds scheme name as prefix to the configfile
+	beatCfg := schemeMap[c.beatname] + ":" + c.configFile
+	// get collector settings
+	set := getCollectorSettings(beatCfg)
+	// get new collector instance
+	otelcol, _ := otelcol.NewCollector(set)
```

Comment:
I see you created your own collector here. I was experimenting with [testbed](https://github.com/elastic/beats/compare/main...mauri870:testbed-experiment?expand=1) that takes care of this. But I found that it is too convoluted for our use case. I was working on getting it working with otelcol, so I believe it is a simpler and more direct approach.